metricbeat/module/containerd/cpu/cpu.go

```diff
@@ -0,0 +1,62 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package cpu
+
+import (
+	p "github.com/elastic/beats/v7/metricbeat/helper/openmetrics"
+	"github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/mb/parse"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/v1/metrics"
+)
+
+var (
+	// HostParser validates Prometheus URLs
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+	}.Build()
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	// Mapping of state metrics
+	mapping := &prometheus.MetricsMapping{
+		Metrics: map[string]prometheus.MetricMap{
+			"container_cpu_total_nanoseconds":  prometheus.Metric("usage.total.ns"),
+			"container_cpu_user_nanoseconds":   prometheus.Metric("usage.user.ns"),
+			"container_cpu_kernel_nanoseconds": prometheus.Metric("usage.kernel.ns"),
+			"container_per_cpu_nanoseconds":    prometheus.Metric("usage.percpu.ns"),
+			"process_cpu_seconds_total":        prometheus.Metric("system.total"),
+		},
+		Labels: map[string]prometheus.LabelMap{
+			"container_id": p.KeyLabel("id"),
```

Comment:
You are using openmetrics library here, is this intentional? Why not `prometheus.KeyLabel()`?

Reply:
By mistake